pkg/skaffold/jib/jib.go

```diff
@@ -18,13 +18,65 @@ package jib
 
 import (
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha3"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 	"github.com/pkg/errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
 )
 
-func GetDependenciesMaven(_ /*workspace*/ string, _ /*a*/ *v1alpha3.JibMavenArtifact) ([]string, error) {
-	return nil, errors.New("jib maven support is unimplemented")
+// GetDependenciesMaven finds the source dependencies for the given jib-maven artifact.
+// All paths are absolute.
+// TODO(coollog): Add support for multi-module projects.
+func GetDependenciesMaven(workspace string, _ /*a*/ *v1alpha3.JibMavenArtifact, isWindows bool) ([]string, error) {
+	return getDependencies(workspace, "pom.xml", "mvn", "mvnw.cmd", isWindows, []string{"jib:_skaffold-files", "-q"}, "jib-maven")
 }
 
-func GetDependenciesGradle(_ /*workspace*/ string, _ /*a*/ *v1alpha3.JibGradleArtifact) ([]string, error) {
-	return nil, errors.New("jib gradle support is unimplemented")
+// GetDependenciesGradle finds the source dependencies for the given jib-gradle artifact.
+// All paths are absolute.
+// TODO(coollog): Add support for multi-module projects.
+func GetDependenciesGradle(workspace string, _ /*a*/ *v1alpha3.JibGradleArtifact, isWindows bool) ([]string, error) {
+	return getDependencies(workspace, "build.gradle", "gradle", "gradlew.bat", isWindows, []string{"_jibSkaffoldFiles", "-q"}, "jib-gradle")
+}
+
+func exists(workspace string, filename string) bool {
```

Comment:
It would be worth making this an `isDir()` that also tests that the `!info.IsDir()`?